src/source/bcm2835_stream.cpp

```diff
@@ -28,17 +30,86 @@ void Stream::begin(unsigned long baud, int flags)
 	}
 	fcntl(fd, F_SETFL, O_RDWR);
 
-	// Use 8 data bit, no parity and 1 stop bit
+
+	speed_t myBaud;
+	switch ( baud )
+	{
+		case      50:	myBaud =      B50; break;
+		case      75:	myBaud =      B75; break;
+		case     110:	myBaud =     B110; break;
+		case     134:	myBaud =     B134; break;
+		case     150:	myBaud =     B150; break;
+		case     200:	myBaud =     B200; break;
+		case     300:	myBaud =     B300; break;
+		case     600:	myBaud =     B600; break;
+		case    1200:	myBaud =    B1200; break;
+		case    1800:	myBaud =    B1800; break;
+		case    2400:	myBaud =    B2400; break;
+		case    4800:	myBaud =    B4800; break;
+		case    9600:	myBaud =    B9600; break;
+		case   19200:	myBaud =   B19200; break;
+		case   38400:	myBaud =   B38400; break;
+		case   57600:	myBaud =   B57600; break;
+		case  115200:	myBaud =  B115200; break;
+		case  230400:	myBaud =  B230400; break;
+		case  460800:	myBaud =  B460800; break;
+		case  500000:	myBaud =  B500000; break;
+		case  576000:	myBaud =  B576000; break;
+		case  921600:	myBaud =  B921600; break;
+		case 1000000:	myBaud = B1000000; break;
+		case 1152000:	myBaud = B1152000; break;
+		case 1500000:	myBaud = B1500000; break;
+		case 2000000:	myBaud = B2000000; break;
+		case 2500000:	myBaud = B2500000; break;
+		case 3000000:	myBaud = B3000000; break;
+		case 3500000:	myBaud = B3500000; break;
+		case 4000000:	myBaud = B4000000; break;
+
+		default:
+			printf("[ERROR] UART invalid baud: %ld for port: %s\n", baud, port);
+			return;
+	}
+
+
+	struct termios options;
+	memset( &options, 0, sizeof(options) );

```

Comment:
```cpp
termios options{};
```
?

Reply:
This is fine and will change to your suggestion as it is quicker than memset. I cannot remember which condition or syntax where having ... = {0} may not necessarily set everything to 0 but memset will always make my code right.

Reply:
Yes the idea is to call the default constructor which of course does not exist as this is a C style struct. Technically it would assign the default values but as that is not a concept in C, the compiler will fall back to assign `0` for all the members.
The stackoverflow discussion bolds that none of the three options sets all members to 0, which looking at the struct definition is by design and so the confusion he has does not apply in our case.
I am fairly confident that calling the default constructor will set everything to 0 and will be a cleaner approach. If Mr Torvalds ever starts adopting C++ into the kernel and they add default values then I think we can reassess this =)

Reply:
Hi. Sorry, I was busy and I also did not have my computer over the weekend.

I can change to `termios options{};` if you want me to. I am getting confused by this. It is because many are mixing C/C++ in their posts on StackOverflow. I do not fully follow you but I think you are saying it is correct because it is a C construct.

Anyway, I have looked at

https://stackoverflow.com/questions/1069621/are-members-of-a-c-struct-initialized-to-0-by-default
https://stackoverflow.com/questions/706030/are-uninitialized-struct-members-always-set-to-zero

and what you mentioned about 
https://stackoverflow.com/questions/61240589/how-to-initialize-a-struct-to-0-in-c

does fall in my understanding. The thing is, our file where this code resides is a .cc file and C++ compiler will be used so in my head, I would like to use `termios options{};` but I am not 100% certain it will set to 0. 

I can do a different pull requests to change to your form if you wanted me to.